### PR TITLE
Pass federation id on pay invoice

### PR DESCRIPTION
**Pull request by okjodom**
Pass federation id on user client request to pay invoice

### CODE REVIEW

**Review by dpc — APPROVED** [review]

File: client/client-lib/src/lib.rs [hunk 1]
```diff
@@ -90,6 +91,18 @@ pub struct PaymentParameters {
...
+pub struct FederationId(pub String);
+
+impl FromStr for FederationId {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        Ok(FederationId(s.to_string()))
+    }
+}
+
```
Comment (elsirion) [inline client/client-lib/src/lib.rs:105]: We already have this in the gateway code, right? Can we remove it there if it's now in the client lib?

**Review by elsirion — COMMENTED** [review]
Needs a rebase and maybe a test fix (didn't look into it)

**Comment by okjodom** [comment]
> Needs a rebase and maybe a test fix (didn't look into it)

Validated the failing `latency test` locally and it's passing. Rebased. Now let's see how [this CI run](https://github.com/fedimint/fedimint/actions/runs/3347750335/jobs/5546074956) goes

**Comment by okjodom** [comment]
> > Needs a rebase and maybe a test fix (didn't look into it)
> 
> Validated the failing `latency test` locally and it's passing. Rebased. Now let's see how [this CI run](https://github.com/fedimint/fedimint/actions/runs/3347750335/jobs/5546074956) goes

Found the bug!

**Review by elsirion — APPROVED** [review]

**Pull request merged**